### PR TITLE
make MigrationObjects optional

### PR DESCRIPTION
**Pull request by alazier**
@segiddins

### CODE REVIEW

**Comment by segiddins** [comment]
Do you need to update the examples?

File: RealmSwift.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -213,7 +213,6 @@
 		0253CD9A1AA0016300AF9E6C /* ObjectSchemaInitializationTests.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; name = ObjectSchemaInitializationTests.swift; path = RealmSwift/Tests/ObjectSchemaInitializationTests.swift; sourceTree = SOURCE_ROOT; };
 		02616BD41A817FC8009919CB /* MigrationTests.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; name = MigrationTests.swift; path = RealmSwift/Tests/MigrationTests.swift; sourceTree = SOURCE_ROOT; };
 		02B822821A532100007F28A5 /* RealmSwift iOS Tests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = "RealmSwift iOS Tests.xctest"; sourceTree = BUILT_PRODUCTS_DIR; };
-		02E334C21A5F3C45009F8810 /* module.modulemap */ = {isa = PBXFileReference; lastKnownFileType = "sourcecode.module-map"; path = module.modulemap; sourceTree = "<group>"; };
```
Comment (segiddins) [inline RealmSwift.xcodeproj/project.pbxproj]: are these changes intentional?

Comment (alazier) [inline RealmSwift.xcodeproj/project.pbxproj]: yes. these files are not needed here.

**Comment by segiddins** [comment]
:+1:
